packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift

```diff
@@ -430,6 +465,131 @@ public class SwiftAmplifyDataStorePlugin: NSObject, FlutterPlugin {
             }
         }
     }
+
+    // Convert the register custom type schemas following the dependencies order
+    // and then convert and register custom type schemas that don't any dependencies 
+    private func registerCustomTypeSchemas(
+        serializedCustomTypeSchemas: [[String: Any]],
+        customTypeSchemaDependenciesOrder: [String]
+    ) throws {
+        for schemaName in customTypeSchemaDependenciesOrder {
+            guard let serializedSchema = serializedCustomTypeSchemas.first(where: { $0["name"] as? String == schemaName } ) else {
+                throw DataStoreError.internalOperation(
+                    "Cannot find serialized custom type schema for the given custom type name.",
+                    "This should not happen, please open an issue at https://github.com/aws-amplify/amplify-flutter/issues"
+                )
+            }
+            let schema: ModelSchema = try FlutterModelSchema.init(serializedData: serializedSchema)
+                .convertToNativeModelSchema(customTypeSchemasRegistry: customTypeSchemaRegistry)
+            customTypeSchemaRegistry.addModelSchema(modelName: schemaName, modelSchema: schema)
+        }
+
+        for serializedCustomTypeSchema in serializedCustomTypeSchemas {
+            guard let schemaName = serializedCustomTypeSchema["name"] as? String else {
+                throw DataStoreError.decodingError(
+                    "Cannot get schema name.",
+                    "Please ensure generating correct models using Ampliy CLi. If this still happens after regenerating models,"
+                        + "please open an issue at https://github.com/aws-amplify/amplify-flutter/issues"
+                )
+            }
+            if (!customTypeSchemaDependenciesOrder.contains(schemaName)) {
+                let schema: ModelSchema = try FlutterModelSchema.init(serializedData: serializedCustomTypeSchema)
+                    .convertToNativeModelSchema(customTypeSchemasRegistry: customTypeSchemaRegistry)
+                customTypeSchemaRegistry.addModelSchema(modelName: schemaName, modelSchema: schema)
+            }
+        }
+    }
+
+    // Resolve custom type dependencies order, the most dependent schema will be registered first
+    // Cicular depdencies are not allowed
+    // The resolved order (array of CustomType names) contains only types that on the both ends of a
+    // dependent relationship
```

Comment:
Not sure if I captured intent correctly but there were some typos here
```suggestion
    // Resolve custom type dependencies order. The most dependent schema will be registered first
    // Circular dependencies are not allowed
    // The resolved order (array of CustomType names) contains only types that are on both ends of a
    // dependent relationship
```